packages/adblocker-webextension/test/index.test.ts

```diff
@@ -125,3 +131,209 @@ describe('#fromWebRequestDetails', () => {
+      }
+
+      public disconnect(): void {
+        this.status = 'disconnected';
+        this.ondata = this._ondata;
+        this.onstop = this._noop;
+      }
+
+      public write(data: ArrayBuffer | Uint8Array): void {
+        const next = new Uint8Array(this._pulled.byteLength + data.byteLength);
+        next.set(this._pulled);
+        next.set(new Uint8Array(data), this._pulled.byteLength);
+        this._pulled = next;
+      }
+
+      public _ondata(event: WebRequest.StreamFilterEventData) {
+        this.write(event.data);
+      }
+
+      public _noop() {}
+
+      public _push(data: ArrayBuffer | Uint8Array, isLast: boolean = false) {
+        if (this.status !== 'transferringdata' && this.status !== 'uninitialized') {
+          this.error = 'Further data transfer cannot be done since the stream is already closed!';
+          this.onerror({
+            data: Uint8Array.from([]),
+          });
+          return;
+        }
+
+        if (this._pushed === 0) {
+          this.status = 'transferringdata';
+        }
+
+        this.ondata({ data });
+        if (isLast === true) {
+          this.onstop({ data: Uint8Array.from([]) });
+          this.status = 'finishedtransferringdata';
+        }
+      }
+
+      public _reset() {
+        this._pulled = new Uint8Array();
+        this._pushed = 0;
+        this.status = 'uninitialized';
+        this.error = '';
+        this.ondata = this._ondata;
+        this.onstart = this._noop;
+        this.onstop = this._ondata;
+        this.onerror = this._noop;
+      }
+    }
+
+    it('stops processing more than MAXIMUM_RESPONSE_BUFFER_SIZE', () => {
+      const streamFilter = new StreamFilter();
+      const details: OnHeadersReceivedDetailsType = {
+        responseHeaders: [
+          {
+            name: 'content-type',
+            value: 'text/text',
+          },
+        ],
+        url: 'https://foo.com/script.js',
+        type: 'main_frame',
+        originUrl: 'https://foo.com/',
+        tabId: 0,
+        requestId: 'req-00',
+      };
+      const request = fromWebRequestDetails(details);
+
+      function filterResponseData(_requestId: string): WebRequest.StreamFilter {
+        return streamFilter;
+      }
+
+      filterRequestHTML(filterResponseData, request, [['replace', [new RegExp('a', 'g'), 'b']]]);
+
+      // Fill MAXIMUM_RESPONSE_BUFFER_SIZE
+      const NOISE_BUFFER_SIZE = 1024 * 1024;
+      for (
+        let i = 0, noise = Uint8Array.from(new Array(NOISE_BUFFER_SIZE).fill('a'.charCodeAt(0)));
+        i < MAXIMUM_RESPONSE_BUFFER_SIZE;
+        i += NOISE_BUFFER_SIZE
+      ) {
+        streamFilter._push(noise);
+      }
+
+      streamFilter._push(Uint8Array.from(['b'.charCodeAt(0)]), true);
+
+      let sum = 0;
+      for (const octet of streamFilter._pulled) {
+        sum += octet;
+      }
+
+      streamFilter._reset();
+
+      expect(sum).to.be.eql('b'.charCodeAt(0) * (MAXIMUM_RESPONSE_BUFFER_SIZE + 1));
+    });
+  });
+});
```

Comment:
if this test is meant to prove that if content-length exceeds the MAX then no html filtering is executed, then we should add another then proves the html filters works if the content-length is just below the MAX

Reply:
We're already doing this, I updated the check to the following to make this more verbose:

```js
// We'll have NOISE_BUFFER_SIZE length of chunks which is replaced from letter 'a' to letter 'b'
// Then one letter 'c' which should not be replaced to letter 'a' or anything
expect(sum).to.be.eql('b'.charCodeAt(0) * MAXIMUM_RESPONSE_BUFFER_SIZE + 'c'.charCodeAt(0));
```

Reply:
Just an idea, we can check this more strictly by dividing the sum into MAXIMUM_RESPONSE_BUFFER_SIZE to check the length of output noises. I don't think this is worth here but if we're going to add more cases, additional check is preferred.